urllib3/connection.py

```diff
@@ -299,7 +302,16 @@ def connect(self):
                     'for details.)'.format(hostname)),
                     SubjectAltNameWarning
                 )
-            match_hostname(cert, self.assert_hostname or hostname)
+            try:
+                asserted_hostname = self.assert_hostname or hostname
```

Comment:
A minor style nit, but I'd rather see this outside the `try` block. While this *shouldn't* be able to throw an exception that causes a problem, there is an attribute access in here that could do all kinds of wacky stuff. Best to keep `try` block scope small. =)